Integrations/SecBI/SecBI.py

```diff
@@ -0,0 +1,197 @@
+        for field in SECBI_INCIDENT_FIELDS:
+            setattr(self, field, incident_json[field])
+
+    def __str__(self) -> str:
+        return json.dumps(self.__dict__)
+
+    def __repr__(self) -> str:
+        return '{} ({})'.format(self.__class__.__name__, self.__str__())
+
+
+class SecBIClient(BaseClient):
+
+    def secbi_get_incidents_list(self, query, limit):
+        """
+        SecBI Get All Incident IDs
+        :param query: The Query by which to filter the Incident IDs
+        :param limit: Limit amount of IDs to return (-1) for all
+        :return: SecBI Incident IDs List
+        """
+        params = get_dict_without_none(query=query, limit=limit)
+        LOG('Performing SecBI get_incidents_list request to {}/{} with params={}'.format(self._base_url,
+                                                                                         SECBI_API_ENDPOINT_INCIDENTS,
+                                                                                         str(params)))
+        try:
+            ids_list_str = self._http_request(method='GET',
+                                              url_suffix=SECBI_API_ENDPOINT_INCIDENTS,
+                                              params=params,
+                                              resp_type='text')
+        except Exception as e:
+            LOG("Error when fetching Incidents list from SecBI: {}".format(e.args[0]))
+            LOG.print_log()
+            raise
+        return json.loads(ids_list_str)
+
+    def secbi_get_incident(self, incident_id):
+        """
+        Get a specific SecBI Incident by SecBI Incident ID
+        :param incident_id: SecBI incident ID
+        :return: A dictionary representation of the SecBI Incident requested
+        """
+        params = get_dict_without_none(incidentId=incident_id, fields=SECBI_INCIDENT_FIELDS)
+        try:
+            raw_incident = self._http_request(method='GET',
+                                              url_suffix='{}/{}'.format(SECBI_API_ENDPOINT_INCIDENT, incident_id),
+                                              params=params)
+            secbi_incident = SecBIIncident(raw_incident)
+            return vars(secbi_incident)
+        except Exception as e:
+            LOG("Error when fetching Incident {} from SecBI: {}".format(incident_id, e.args[0]))
+            LOG.print_log()
+            raise
+
+    def secbi_get_incident_by_host(self, host):
+        """
+        Get a specific SecBI Incident by Host
+        :param host: The host by which to get a SecBI Incident
+        :return:  A dictionary representation of the SecBI Incident found to contain the supplied host
+        """
+        params = get_dict_without_none(host=host, fields=SECBI_INCIDENT_FIELDS)
+        try:
+            raw_incident = self._http_request(method='GET', url_suffix=SECBI_API_ENDPOINT_INCIDENT, params=params)
+            secbi_incident = SecBIIncident(raw_incident)
+            return vars(secbi_incident)
```

Comment:
I think this is a bit of an overkill here, why not init a dictionary and map the incident fields to it?